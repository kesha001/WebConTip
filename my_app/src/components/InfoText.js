import React, { Component } from "react";

class InfoText extends Component {
	constructor(props) {
		super(props);
		this.state = {
			data: [],
			loaded: false,
			placeholder: "Loading"
		};
	}


	componentDidMount() {
		const token = localStorage.getItem('jwt access');
        console.log(token);
        const options = {
            method: "GET",
            headers: {
                'Accept': 'application/json',
				'Content-Type': 'application/json',
				'Authorization': `JWT ${token}`
            }
        }
		fetch("http://localhost:8000/api/v1/account/users/me/", options)
			.then(response => {
				console.log(response);
				if (response.status > 400) {
					return this.setState(() => {
						return { placeholder: "Something went wrong!" };
					});
				}
				return response.json();
			})
			.then(data => {
				console.log(data);
				this.setState(() => {
					return {
						data,
						loaded: true
					};
				});
			});
	}

	render() {
		return (
			<div className="profile_info">
				<p>Username: {this.state.data.username} </p>
				<p>Email: {this.state.data.email} </p>
			</div>
		);
	}
}

export default InfoText;
