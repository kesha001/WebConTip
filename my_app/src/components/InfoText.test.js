import React from 'react';
import { render } from '@testing-library/react';
import ReactDOM from 'react-dom';
import InfoText from './InfoText';
import {shallow, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });


it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<InfoText />, div);
  ReactDOM.unmountComponentAtNode(div);
});

describe('HeaderComponent', () => {
    
    const props = {
        data: [],
        loaded: false,
        placeholder: "Loading"
    };

    const component = shallow(<InfoText {...props} />)

    it('renders properly', () => {
        expect(component).toMatchSnapshot()
      })

    it('to be no token', () => {
        expect(component.prop('Token')).toEqual(undefined)
     })

    it('username', () => {
    expect(component.find('.username_prof')).toHaveLength(0)
    })

    it('email', () => {
        expect(component.find('.email_prof')).toHaveLength(0)
     })
    
});