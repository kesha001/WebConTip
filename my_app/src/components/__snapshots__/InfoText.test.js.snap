// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderComponent renders properly 1`] = `ShallowWrapper {}`;
